hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonOutputCommitter.java

```diff
@@ -336,9 +336,10 @@ private String overwritePartitions(CarbonLoadModel loadModel, LoadMetadataDetail
           .filter(partitionList::contains).collect(Collectors.toList());
       if (!overlappingPartitions.isEmpty()) {
         List<LoadMetadataDetails> validLoadMetadataDetails =
-            loadModel.getLoadMetadataDetails().stream().filter(
-                loadMetadataDetail -> !loadMetadataDetail.getLoadName()
-                    .equalsIgnoreCase(newMetaEntry.getLoadName())).collect(Collectors.toList());
+            loadModel.getLoadMetadataDetails().stream().filter(loadMetadataDetail ->
+                !loadMetadataDetail.getLoadName().equalsIgnoreCase(newMetaEntry.getLoadName())
+                    && !loadMetadataDetail.getSegmentStatus()
+                    .equals(SegmentStatus.MARKED_FOR_DELETE)).collect(Collectors.toList());
```

Comment:
We should filter out segments which has segment state as SUCCESS, otherwise the same scenario after compaction will fail

Reply:
thanks for pointing out, done